stdlib/LinearAlgebra/src/svd.jl

```diff
@@ -196,6 +196,8 @@ function ldiv!(A::SVD{T}, B::StridedVecOrMat) where T
     view(A.Vt,1:k,:)' * (view(A.S,1:k) .\ (view(A.U,:,1:k)' * B))
 end
 
+inv(F::SVD) = (F.S .\ F.Vt)' * F.U'
```

Comment:
Thanks for adding this. We shouldn't probably check for exact zeros in `S` and throw a `SingularException` to be consistent with the other `Factorizations`.

Reply:
I assume you meant we _should_?
```julia
function Base.inv(F::SVD)
    for i in eachindex(F.S)
        @inbounds iszero(F.S[i]) && throw(SingularException(i))
    end
    (F.S .\ F.Vt)' * F.U'
end
```

Reply:
Numbers in `F.S` are ordered, so should we start checking from the end? In fact, if `!iszero(F.S[end])` then all other singular values should be >0 and we're all good, otherwise we could throw a `SingularException(length(F.S))`? Something like this...

Reply:
Good point. However, I guess the question is if we also want to throw a `SingularException` if the singular values are not ordered; one might manually construct a `SVD` where the entries in `S` are not in descending order. A case that comes to mind is, for example, the inverse factorization of an `SVD`. Here one simply inverts each singular value and the order is reversed.

Perhaps the fact that the singular values are in descending order is a property of `svd` rather than `SVD` and we shouldn't rely on it?

Reply:
You're absolutely right... which makes me think that this "inverse factorization" approach without permutation/ordering is not so good anyway. If some stdlib function returned an `SVD`, I think one should rely on the ordering of the singular values. That's how "everybody" is used to interpret an SVD.

That is not to say we should rely on it here. That little loop is not the bottleneck anyway.

Reply:
Alright, let's keep the simple loop then.